compiler/rustc_interface/src/passes.rs

```diff
@@ -301,8 +301,41 @@ fn early_lint_checks(tcx: TyCtxt<'_>, (): ()) {
         for (ident, mut spans) in identifiers.drain(..) {
             spans.sort();
             if ident == sym::ferris {
+                enum FerrisFix {
+                    SnakeCase,
+                    ScreamingSnakeCase,
+                    PascalCase,
+                }
+
+                impl FerrisFix {
+                    const fn as_str(self) -> &'static str {
+                        match self {
+                            FerrisFix::SnakeCase => "ferris",
+                            FerrisFix::ScreamingSnakeCase => "FERRIS",
+                            FerrisFix::PascalCase => "Ferris",
+                        }
+                    }
+                }
```

Comment:
This doesn't really need to be an enum, it's used for a single purpose so constants would have been fine and more compact. Or even literals considering `SnakeCase` and `PascalCase` are each constructed once and `ScreamingSnakeCase` only twice.

(but already approved so 🤷 )